src/gtest/test_mempool.cpp

```diff
@@ -134,13 +134,27 @@ TEST(Mempool, TxInputLimit) {
     // The -mempooltxinputlimit check doesn't set a reason
     EXPECT_EQ(state3.GetRejectReason(), "");
 
+    // Resize the transaction, add CSW inputs
+    mtx.vin.resize(10);
+    mtx.vcsw_ccin.resize(1);
+
+    // Check it now fails due to exceeding the total inputs limit
+    CValidationState state3csw;
+    CTransaction txWithCsw(mtx);
+    EXPECT_FALSE(AcceptTxToMemoryPool(pool, state3csw, txWithCsw, false, &missingInputs));
+    // The -mempooltxinputlimit check doesn't set a reason
+    EXPECT_EQ(state3csw.GetRejectReason(), "");
+
     // Clear the limit
     mapArgs.erase("-mempooltxinputlimit");
 
     // Check it no longer fails due to exceeding the limit
     CValidationState state4;
     EXPECT_FALSE(AcceptTxToMemoryPool(pool, state4, tx3, false, &missingInputs));
     EXPECT_EQ(state4.GetRejectReason(), "bad-txns-version-too-low");
+    CValidationState state4csw;
+    EXPECT_FALSE(AcceptTxToMemoryPool(pool, state4csw, txWithCsw, false, &missingInputs));
+    EXPECT_EQ(state4.GetRejectReason(), "bad-txns-version-too-low");
```

Comment:
I would be great to provide a positive case, where a tx with csw is accepted to mempool. This is especially important if we cannot check the reason for rejection (how are you sure it's not rejected for an unforeseen reason?)